File: blog17_removeAdjacentDuplicates.rb
def remove_duplicates(s)
    # create an empty array
    stack = []
    # iterate over input string
    s.each_char do |item|
        # remove the last element of the stack if it
        # is equal to the element of the string
      item == stack[-1] ? stack.pop : stack.push(item) 
    end
    stack.join
end

File: blog17_removeAdjacentDuplicates.js
removeDuplicates = S => {
    // create an empty array
    let stack = []
    // iterate over input string
    for (let i = 0; i < S.length; i ++) {
        // check if the element of the string 
        // is equal to the last element of the stack
      if (S[i] == stack[stack.length - 1]) {
        // if yes, remove the last element of the stack
        stack.pop()
      } else {
        // if not, put the element on the stack
        stack.push(S[i])
      }
    }
    return stack.join("")
}
